src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -338,11 +339,21 @@ internal class GroupBatchCheckIterator(
 ) : Iterator<StoredGroupedMessageBatch> by original {
     override fun next(): StoredGroupedMessageBatch = original.next().also { batch ->
         if (batch.messages.size > 1) {
-            batch.messages.reduce { first, second ->
-                check(first.timestamp <= second.timestamp) {
-                    "Unordered message received for: ${batch.toShortInfo()} batch, between ${first.toShortInfo()} and ${second.toShortInfo()} messages"
+            val map = hashMapOf<Stream, StreamMetadata>()
```

Comment:
We are chasing better performance but yet we add extra checks during batch processing. I hope collecting all data in hashmap won't have much effect on it